scrapy/http/request/form.py

```diff
@@ -48,7 +49,11 @@ def from_response(cls, response, formname=None, formid=None, formnumber=0, formd
         form = _get_form(response, formname, formid, formnumber, formxpath)
         formdata = _get_inputs(form, formdata, dont_click, clickdata, response)
         url = _get_form_url(form, kwargs.pop('url', None))
-        method = kwargs.pop('method', form.method)
+
+        method = kwargs.pop('method', form.method).upper()
```

Comment:
I think previously passing method=None explicitly was supported, but now it is not - could you please fix this small backwards compatibility issue?

Reply:
and `method=None` should be `GET` or `None`?

Reply:
`GET`, I believe.

Reply:
That's an interesting question, because current behavior is POST. 

1) method is not passed => we use form.method fallback
2) `<form method>` not defined in HTML => I believe lxml returns "GET" in this case (need to check it), so method becomes GET
3) method=None is passed explicitly => currently method=None is passed through to `__init__`, which means method becomes POST. 

(3) is what is changed - a case where an user pass method=None explicitly. Not sure about the use case, just noticed that it is backwards incompatible. Passing through None should be enough to fix it.